### ISSUE TITLE
workflow to extract text

### ISSUE THREAD

**Issue by eflister** [Yuras/pdf-toolbox#62]
hi - i just used  #master to do the common thing of extracting all text from a pdf.  it worked, thanks for the nice library!  it took a while to figure out how to do it and required more contortions than i expected.  perhaps you could add some api support for such a basic task?  here's what i wound up with, is this what you expect users to do?

```haskell
main = do
  withPdfFile "file.pdf" $ \pdf -> do
    txt <- extract pdf =<< catalogPageNode =<< documentCatalog =<< document pdf

extract pdf = (T.concat <$>) . (traverse ((extract' =<<) . loadPageNode pdf) =<<) . pageNodeKids
  where
    extract' (PageTreeLeaf tn) = pageExtractText tn
    extract' (PageTreeNode tn) = extract pdf tn
```

**Comment by Yuras** [comment]
Yeah, simpler API would be great. Though I'm not sure how exactly it should look like. I'll think about it, thank you for the input.

**Comment by olafklinke** [comment]
Actually, the [pdftotext](https://hackage.haskell.org/package/pdftotext) package works quite nicely for that purpose. One should not bloat this nice little library with all sorts of use cases.
